controllers/imagerepository_controller.go

```diff
@@ -122,12 +137,17 @@ func (r *ImageRepositoryReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 	ok, when := r.shouldScan(imageRepo, reconcileStart)
 	if ok {
 		reconcileErr := r.scan(ctx, &imageRepo, ref)
-		if err = r.Status().Update(ctx, &imageRepo); err != nil {
+		if err := r.Status().Update(ctx, &imageRepo); err != nil {
 			return ctrl.Result{Requeue: true}, err
 		}
 		if reconcileErr != nil {
+			r.event(imageRepo, events.EventSeverityError, reconcileErr.Error())
 			return ctrl.Result{Requeue: true}, reconcileErr
 		}
+		// emit successful scan event
+		if rc := apimeta.FindStatusCondition(imageRepo.Status.Conditions, meta.ReconciliationSucceededReason); rc != nil {
```

Comment:
If this isn't straight-forwardly contingent on getting a `nil` error return, I would consider returning a bool saying whether a scan was done or not, to avoid this indirect way of finding out.

Reply:
By the time we reach this, a scan was most definitely completed.

`rc` is not of error type, it's `*metav1.Condition`, can be `nil` if no condition was found (should never happen). This is used just to get the condition `Message` used as the event message. Does that make any sense?

Reply:
OIC, I didn't notice the message was used. OK, that's fair enough!